add-on/panels/js/roomToc.jsx

```diff
@@ -12,51 +13,71 @@ loop.roomToc = (function() {
   function init() {
     loop.shared.utils.getBoolPreference = function foo() {};
 
-    var dispatcher = new loop.Dispatcher();
-    var sdkDriver = new loop.OTSdkDriver({
-      // For the standalone, always request data channels. If they aren't
-      // implemented on the client, there won't be a similar message to us, and
-      // we won't display the UI.
-      constants: {},
-      useDataChannels: true,
-      dispatcher: dispatcher,
-      sdk: OT
-    });
+    var requests = [
+      ["GetAllConstants"],
+      ["GetAllStrings"],
+      ["GetLocale"]
+    ];
 
-    var activeRoomStore = new loop.store.ActiveRoomStore(dispatcher, {
-      sdkDriver: sdkDriver
-    });
+    loop.requestMulti.apply(null, requests).then(function(results) {
+      let [constants, stringBundle, locale] = results;
+      mozL10n.initialize({
+        locale: locale,
+        getStrings: function(key) {
+          if (!(key in stringBundle)) {
+            console.error("No string found for key: ", key);
+            return JSON.stringify({ textContent: "foo" });
+          }
 
-    // XXX akita: Get loop constants
-    var roomStore = new loop.store.RoomStore(dispatcher, {
-      constants: { CONSTANT: false }
-    });
+          return JSON.stringify({ textContent: stringBundle[key] });
+        }
+      });
 
-    loop.store.StoreMixin.register({
-      activeRoomStore: activeRoomStore,
-      roomStore: roomStore
-    });
+      var dispatcher = new loop.Dispatcher();
+      var sdkDriver = new loop.OTSdkDriver({
+        // For the standalone, always request data channels. If they aren't
+        // implemented on the client, there won't be a similar message to us, and
+        // we won't display the UI.
+        constants: {},
+        useDataChannels: true,
+        dispatcher: dispatcher,
+        sdk: OT
+      });
 
-    window.addEventListener("unload", function() {
-      dispatcher.dispatch(new sharedActions.WindowUnload());
-    });
+      var activeRoomStore = new loop.store.ActiveRoomStore(dispatcher, {
+        sdkDriver: sdkDriver
+      });
 
-    ReactDOM.render(<tocViews.TableOfContentView
-                      activeRoomStore={activeRoomStore}
-                      dispatcher={dispatcher}
-                      isScreenShareActive={false} />, document.querySelector("#main"));
+      var roomStore = new loop.store.RoomStore(dispatcher, {
+        constants: constants
+      });
 
-    var locationData = sharedUtils.locationData();
-    var hash = locationData.hash.match(/#(.*)/);
+      loop.store.StoreMixin.register({
+        activeRoomStore: activeRoomStore,
+        roomStore: roomStore
```

Comment:
Fyi, with es6 you can leave off the value when the value is using a variable with the name of the key, e.g., `{ activeRoomStore, roomStore }` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#Property_definitions